components/base/wheeled/sensorbase.go

```diff
@@ -355,10 +362,72 @@ func (sb *sensorBase) SetVelocity(
 	ctx context.Context, linear, angular r3.Vector, extra map[string]interface{},
 ) error {
 	sb.opMgr.CancelRunning(ctx)
-	sb.setPolling(false)
+	// check if a sensor context has been started
+	if sb.sensorLoopDone != nil {
+		sb.sensorLoopDone()
+	}
+
+	sb.setPolling(true)
+	// start a sensor context for the sensor loop based on the longstanding base
+	// creator context, and add a timeout for the context
+	timeOut := time.Duration(10 * time.Second)
+	var sensorCtx context.Context
+	sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
+
+	if sb.velocitiesSensor != nil {
+		sb.logger.Warn("not using sensor for SetVelocityfeedback, this feature will be implemented soon")
+		// TODO implement control loop here instead of placeholder sensor pllling function
+		sb.pollsensors(sensorCtx, extra)
+		return errors.New("setvelocity with sensor feedback not currently implemented, remove movement sensor reporting linear and angular velocity ")
```

Comment:
q for reviewers: should I log an error rather than return it here and allow someone to use `SetVelocity`? This base won't use the underlying wheeled base's `SetVelocity` function always? Right now, the `sensorbase` can only access SetVelocity it if the sensor provided is an orientation reporting sensor.